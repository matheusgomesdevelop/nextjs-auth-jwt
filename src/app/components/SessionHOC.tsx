'use client';

import React, { ComponentType, useEffect } from 'react';

import { useRouter } from 'next/navigation';

import { AuthUserSchema } from '@/schemas/AuthSchema';

import useSession from '@/app/hooks/useSession';

export interface SessionHOCProps {
    data: {
        session: AuthUserSchema | undefined;
    };
    error: boolean;
    loading: boolean;
}

export const withSessionHOC = <P extends object>(Component: ComponentType<P & SessionHOCProps>): React.FC<P> => {
    const Wrapper: React.FC<P> = (props) => {
        const { data, loading, error } = useSession();

        const router = useRouter();

        useEffect(() => {
            if (!loading && error) {
                if (window.location.pathname === '/') {
                    router.push('/');
                } else {
                    router.push('/unauthorized');
                }
            } else if (!loading && !error && data.session && window.location.pathname === '/') {
                router.push('/welcome');
            }
        }, [loading, error, router, data.session]);

        const modifiedProps = {
            ...props,
            data,
            error,
            loading,
        };

        return <Component {...modifiedProps} />;
    };

    return Wrapper;
};
