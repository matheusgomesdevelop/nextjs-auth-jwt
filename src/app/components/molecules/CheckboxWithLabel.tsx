import React from 'react';

import BaseField from '@/core/components/Field/Field';
import Checkbox, { CheckboxProps } from '../atoms/Checkbox';
import Label from '../atoms/Label';

interface CheckboxWithLabelProps extends CheckboxProps {
    labelText: string;
    name: string;
}

const CheckboxWithLabel: React.FC<CheckboxWithLabelProps> = ({ labelText, name, ...props }) => (
    <div className="items-center flex gap-6">
        <BaseField render={<Checkbox id="checkbox" {...props} />} name={name} type="checkbox" />

        <Label
            onClick={() => {
                const get_checkbox = document.getElementById('checkbox');
                get_checkbox && get_checkbox.click();
            }}
        >
            {labelText}
        </Label>
    </div>
);

export default CheckboxWithLabel;
