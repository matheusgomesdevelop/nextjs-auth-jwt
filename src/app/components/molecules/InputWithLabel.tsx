import React from 'react';

import BaseField from '../templates/BaseField';
import Label from '../atoms/Label';
import Input, { InputProps } from '../atoms/Input';

interface InputWithLabelProps extends InputProps {
    labelText: string;
    type: React.HTMLInputTypeAttribute;
    name: string;
    id?: string;
}

const InputWithLabel: React.FC<InputWithLabelProps> = ({ labelText, type, name, id, placeholder, ...props }) => (
    <div className="flex-col flex gap-6">
        <Label className="font-medium text-sm" htmlFor={name}>
            {labelText}
        </Label>

        <BaseField
            render={<Input placeholder={placeholder} type={type} name={name} id={id ?? name} {...props} />}
            name={name}
            type={type}
        />
    </div>
);

export default InputWithLabel;
