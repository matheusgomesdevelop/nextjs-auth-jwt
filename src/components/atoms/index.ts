import Checkbox from './Checkbox/Checkbox';
import Button, { ButtonRef } from './Button/Button';
import Avatar from './Avatar/Avatar';
import Label from './Label/Label';
import Input from './Input/Input';
import Title from './Title/Title';
import Text from './Text/Text';
import Icon from './Icon/Icon';
import Link from './Link/Link';

export { Checkbox, Button, Avatar, Title, Label, Input, Text, Icon, Link };

export type { ButtonRef };
