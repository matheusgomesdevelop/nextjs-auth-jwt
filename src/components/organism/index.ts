import BlockUI from './BlockUI/BlockUI';
import Footer from './Footer/Footer';
import Header from './Header/Header';

export { BlockUI, Footer, Header };
