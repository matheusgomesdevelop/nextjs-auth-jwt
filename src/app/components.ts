// Atoms
import {
  AAvatar,
  ATitle,
  A_Label,
  A_Input,
  ACheckbox,
  A_Button,
} from "@/components/atoms";

// Molecules
import { M_InputWithLabel } from "@/components/molecules";

export {
  AAvatar,
  ATitle,
  A_Label,
  A_Input,
  ACheckbox,
  A_Button,
  M_InputWithLabel,
};
