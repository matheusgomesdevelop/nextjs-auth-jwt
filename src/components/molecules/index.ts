import CheckboxWithLabel from './CheckboxWithLabel/CheckboxWithLabel';
import InputWithLabel from './InputWithLabel/InputWithLabel';
import Dialog from '../../core/components/Dialog/Dialog';
import Toast from '../../core/components/Toast/Toast';

export { CheckboxWithLabel, InputWithLabel, Dialog, Toast };
