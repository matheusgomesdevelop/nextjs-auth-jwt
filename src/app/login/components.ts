import { Avatar } from "@/components/atoms";

export { Avatar };
