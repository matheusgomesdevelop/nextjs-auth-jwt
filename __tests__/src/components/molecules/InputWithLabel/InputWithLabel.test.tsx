import React from 'react';

import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import { InputWithLabel } from '@/components/molecules';

import BaseForm from '@/core/components/Form/Form';

describe('Deve renderizar o InputWithLabel corretamente.', () => {
    it('Deve renderizar o InputWithLabel na tela.', () => {
        render(
            <BaseForm initialValues={{}} onSubmit={() => {}}>
                <InputWithLabel labelText="Email" name="email" type="email" />
            </BaseForm>
        );

        const getInputWithLabel = screen.getByTestId('input-with-label-email-testid');

        expect(getInputWithLabel).toBeInTheDocument();
    });

    it('Deve renderizar o Label com o texto.', () => {
        render(
            <BaseForm initialValues={{}} onSubmit={() => {}}>
                <InputWithLabel labelText="Email" name="email" type="email" />
            </BaseForm>
        );

        const getLabel = screen.getByTestId('label-email-testid');

        expect(getLabel).toBeInTheDocument();
        expect(getLabel.textContent).not.toHaveLength(0);
    });

    it('Deve renderizar o campo input.', () => {
        render(
            <BaseForm initialValues={{}} onSubmit={() => {}}>
                <InputWithLabel labelText="Email" name="email" type="email" />
            </BaseForm>
        );

        const getInput = screen.getByTestId('input-email-testid');

        expect(getInput).toBeInTheDocument();
        expect(getInput).toHaveProperty('type');
        expect(getInput).toHaveProperty('name');
    });

    it('Deve preservar a estrutura visual do componente.', () => {
        const getInputWithLabel = renderer
            .create(
                <BaseForm initialValues={{}} onSubmit={() => {}}>
                    <InputWithLabel labelText="Email" name="email" type="email" />
                </BaseForm>
            )
            .toJSON();

        expect(getInputWithLabel).toMatchSnapshot();
    });
});
