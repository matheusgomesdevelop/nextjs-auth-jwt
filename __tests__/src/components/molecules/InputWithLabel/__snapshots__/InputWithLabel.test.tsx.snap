// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar o InputWithLabel corretamente. Deve preservar a estrutura visual do componente. 1`] = `
<form
  action="#"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="input_label"
    data-testid="input-with-label-email-testid"
  >
    <label
      className="label"
      data-testid="label-email-testid"
      htmlFor="email"
    >
      Email
    </label>
    <input
      className="p-inputtext p-component"
      data-pc-name="inputtext"
      data-pc-section="root"
      data-testid="input-email-testid"
      name="email"
      onBeforeInput={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onInput={[Function]}
      onKeyDown={[Function]}
      onPaste={[Function]}
      type="email"
    />
  </div>
</form>
`;
